File: sorting/bubble_sort/bubble_sort.cpp
#include<iostream>

using namespace std;

int main(){
	cout << "Enter number of Elements to sort" << endl;
	int n;
	cin >> n;
	int a[n];
	cout << "Enter Elements" << endl;
	for(int i=0; i<n ;i++){
		cin >> a[i];
	}

	// Bubble Sort begins
	for (int i = 0; i < n-1; i++){
		for (int j = 1; j < n-i; j++){
			if(a[j-1]>a[j]){
				int temp = a[j];
				a[j] = a[j-1];
				a[j-1] = temp;
			}

		}
	}
	// Bubble Sort ends
	cout << "Sorted array is: " ;
	for(int i=0; i<n ;i++){
		cout << a[i] << " " ;
	}
	return 0;
}

File: sorting/bubble_sort/bubble_sort.rb
puts "Number of elements to be sorted"
n = gets.to_i
puts "Enter the elements"
a = []
for i in 0...n
	a[i] = gets.to_i
end

# Bubble sort begins here
for i in 0...n-1
	for j in 1...n-i
		if a[j-1] > a[j]
			temp = a[j]
			a[j] = a[j-1]
			a[j-1] = temp
		end
	end
end
# Bubble sort ends here

puts "Sorted array is: "
for i in 0...n 
	print a[i]
	print " "
end
